analyzer/analyzer.go

```diff
@@ -234,6 +234,9 @@ func (a Applier) ApplyLayers(imageID digest.Digest, layerIDs []string) (types.Im
 		if layer.SchemaVersion == 0 {
 			return types.ImageDetail{}, xerrors.Errorf("layer cache missing: %s", layerID)
 		}
+		if layer.ID == "" {
+			layer.ID = digest.Digest(layerID)
+		}
```

Comment:
```suggestion
		layer.ID = digest.Digest(layerID)
```
`layer.ID` should be always empty. We don't need this `if` condition. Even if it is not empty (it is not supposed to happen), we should overwrite `layer.ID` with `layerID`.